pkg/services/govmomi/service_test.go

```diff
@@ -93,9 +101,136 @@ func Test_reconcilePCIDevices(t *testing.T) {
 			pciDevices := devices.SelectByBackingInfo(&types.VirtualPCIPassthroughDynamicBackingInfo{
 				AllowedDevice: []types.VirtualPCIPassthroughAllowedDevice{
 					{DeviceId: 1234, VendorId: 5678},
-				}})
+				},
+			})
 			g.Expect(pciDevices).To(HaveLen(2))
 			return nil
 		})
 	})
 }
+
+func Test_ReconcileStoragePolicy(t *testing.T) {
+	var vmCtx *virtualMachineContext
+	var g *WithT
+	var vms *VMService
+
+	before := func() {
+		vmCtx = emptyVirtualMachineContext()
+		vmCtx.Client = fake.NewClientBuilder().Build()
+
+		vms = &VMService{}
+	}
+	t.Run("when VM has no storage policy spec", func(t *testing.T) {
+		g = NewWithT(t)
+		before()
+		vmCtx.VSphereVM = &infrav1.VSphereVM{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "vsphereVM1",
+				Namespace: "my-namespace",
+			},
+			Spec: infrav1.VSphereVMSpec{
+				VirtualMachineCloneSpec: infrav1.VirtualMachineCloneSpec{},
+			},
+		}
+		g.Expect(vms.reconcileStoragePolicy(context.Background(), vmCtx)).ToNot(HaveOccurred())
+		g.Expect(vmCtx.VSphereVM.Status.TaskRef).To(BeEmpty())
+	})
+
+	t.Run("when the requested storage policy does not exists should fail", func(t *testing.T) {
+		g = NewWithT(t)
+		before()
+		model, err := storagePolicyModel()
+		g.Expect(err).ToNot(HaveOccurred())
+
+		simulator.Run(func(ctx context.Context, c *vim25.Client) error {
```

Comment:
Optional: I don't know which method is better, this one or the other. Totally fine to keep as is.

Wondering if we should use the vcsim package here instead via:

```go
  simr, err := vcsim.NewBuilder().WithModel(model).Build()
  defer simr.Destroy()
  g.Expect(err).ToNot(gomega.HaveOccurred())

  ...
  // for session setup
  simr.ServerURL().Host
  simr.Username()
  simr.Password()
  ...

  vm, err := getPoweredoffVM(ctx, c)
  ...
```

Reply:
ah yeah, in this case I just tried to keep consistency with the other test. Happy to do any of the ways tho

Reply:
Lets keep as is then 👍